Doc/glossary.rst

```diff
@@ -1138,6 +1138,11 @@ Glossary
       an :term:`expression` or one of several constructs with a keyword, such
       as :keyword:`if`, :keyword:`while` or :keyword:`for`.
 
+   static type checker
```

Comment:
I found this useful to add context to the `warnings` docs. I'll send a separate PR with just this change too so we can backport it.